### PR TITLE
feat: Handle null and empty string values in OpenAPI Enum field

### PR DESCRIPTION
**Pull request by zamanafzal**
__Jira Ticket:__ [ENT-6697](https://2u-internal.atlassian.net/browse/ENT-6697)

__Description:__
We need to handle the null and empty string values in the Enum field for the Open API schema [enterprise-access](https://github.com/edx/enterprise-access) service. A sample implementation is available at the PR [prototype: drf-spectacular by iloveagent57 · Pull Request #89 · openedx/enterprise-access](https://github.com/openedx/enterprise-access/pull/89/commits). 
The file oas.py is taken from https://github.com/openedx/enterprise-access/pull/89/files.

**Acceptance Criteria:**
1. A valid Enterprise Access API.yaml file is created to handle nullable enums and generate GraphQL schema.

### CODE REVIEW

File: enterprise_access/settings/base.py [hunk 1]
```diff
@@ -140,6 +141,14 @@ def root(*path_fragments):
     'TEST_REQUEST_DEFAULT_FORMAT': 'json',
 }
 
+SPECTACULAR_SETTINGS = {
+    'TITLE': 'Enterprise Access API',
+    'DESCRIPTION': 'API for controlling request-based access to enterprise subsidized enrollments.',
+    'VERSION': '1.0.0',
+    'ENUM_ADD_EXPLICIT_BLANK_NULL_CHOICE': False,
+    'POSTPROCESSING_HOOKS': ['enterprise_access.oas.postprocess_schema_enums']
```
Comment (saleem-latif) [inline enterprise_access/settings/base.py:149]: I can't find this file in the PR, I think you forgot to add to git.

File: enterprise_access/settings/base.py [hunk 2]
```diff
@@ -140,6 +141,14 @@ def root(*path_fragments):
     'TEST_REQUEST_DEFAULT_FORMAT': 'json',
 }
 
+SPECTACULAR_SETTINGS = {
+    'TITLE': 'Enterprise Access API',
+    'DESCRIPTION': 'API for controlling request-based access to enterprise subsidized enrollments.',
+    'VERSION': '1.0.0',
+    'ENUM_ADD_EXPLICIT_BLANK_NULL_CHOICE': False,
```
Comment (saleem-latif) [inline enterprise_access/settings/base.py:148]: We do not need this setting, this setting is only useful for `drf_spectacular.hooks.postprocess_schema_enums` hook and we are not using that here.

**Review by saleem-latif — CHANGES_REQUESTED** [review]

File: enterprise_access/apps/core/oas.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+from drf_spectacular.plumbing import ResolvedComponent, list_hash, load_enum_name_overrides, safe_ref, warn
+from drf_spectacular.settings import spectacular_settings
+from inflection import camelize
+from rest_framework.settings import api_settings
+
+
+def postprocess_schema_enums(result, generator, **kwargs):  # pylint: disable=too-many-statements
+    """
+    simple replacement of Enum/Choices that globally share the same name and have
```
Comment (saleem-latif) [inline enterprise_access/apps/core/oas.py:16]: I think we should update the doc string to reflect current behavior.

**Review by saleem-latif — APPROVED** [review]

**Review by johnnagro — COMMENTED** [review]
Please close without merging. Thanks for putting this together. We discussed this in Scrum of Scrums this morning:

https://2u-internal.atlassian.net/wiki/spaces/SOL/pages/300580900/EMET+Scrum+of+Scrums+Notes#2023%2F01%2F31

The decision was that for now we do NOT recommend merging this. Please close without merging. We may come back to it later. For now we'd prefer making ENUM columns non-nullable. Please find @iloveagent57 or myself for further context.

**Pull request closed without merging**